src/content/blog/astro-420.mdx

```diff
@@ -0,0 +1,126 @@
+publishDate: "January 18, 2024"
+authors:
+  - erika
+  - ema
+  - matthew
+  - nate
+  - bjorn
+coverImage: "/src/content/blog/_images/astro-420/post-header-4.2.webp"
+socialImage: "/src/content/blog/_images/astro-420/og-image-4.2.webp"
+lang: "en"
+---
+
+Astro 4.2 is now available! This release includes new experimental features to try out, improvements to accessibility rules, the ability for remark plugins to customize image optimization in Markdown, and many more improvements and bug fixes.
+
+Highlights include:
+
+- **[(Experimental) Support for prerendering pages using the Speculation Rules API](#experimental-prerendering-pages-using-the-speculation-rules-api)**
+- **[(Experimental) Reworked routing priority for injected routes](#experimental-reworked-routing-priority-for-injected-routes)**
+- **[Improved accessibility rules](#improved-accessibility-rules)**
+- **[Customizable image optimization in Markdown](#customizable-image-optimization-in-markdown)**
+
+As a side note, this is the first release of Astro to include no highlights from core team members! This is a huge milestone for us, as it means that Astro is now big enough to have a thriving community of contributors capable of spearheading releases. Thank you to everyone who contributed to this release!
+
+## How to upgrade
+
+To take advantage of the latest features, make sure you're running the latest version of Astro. You can upgrade to Astro 4.2 by running the `@astrojs/upgrade` command:
+
+```sh
+npx @astrojs/upgrade
+```
+
+or by running the upgrade command for your package manager:
+
+```sh
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## (Experimental) Prerendering pages using the Speculation Rules API
+
+Thanks to [Ross Robino](https://github.com/rossrobino), Astro's `prefetch` feature now has experimental support for prerendering pages using the currently Chromium-exclusive [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/API/Speculation_Rules_API). This API allows you to prerender pages on the client, and even run client-side JavaScript on pages that the user is likely to visit next, allowing for a faster browsing experience.
+
+To enable this feature, add the following to your `astro.config.mjs` file:
+
+```diff
+import { defineConfig } from 'astro/config';
+
+// https://astro.build/config
+export default defineConfig({
++    prefetch: true,
++    experimental: {
++        clientPrerender: true,
++    },
+});
+```
+
+This will respect your existing `prefetch` configuration options, but will now prerender all links with the `data-astro-prefetch` attribute on the client, instead. A built-in fallback is provided for browsers that do not yet support the Speculation Rules API.
+
+Additionally, you can use the `document.prerendering` property in `<script>` tags to check whether the page is being prerendered:
+
+```html
+<script>
+		if (document.prerendering) {
+				// prerendering is enabled
+		}
+</script>
```

Comment:
```suggestion
<script>
  if (document.prerendering) {
    // prerendering is enabled
  }
</script>
```
Please adjust this however you prefer, but it just seemed like all those tabs were *really* spacious. Can probably get away with fewer tabs, or those as spaces instead. Do whatever looks right!

![image](https://github.com/withastro/astro.build/assets/5098874/d0296dbd-9189-494d-b9bf-5a1b02afd32e)